File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.java
class Solution {
    public ListNode modifiedList(int[] nums, ListNode head) {
        var hs = new HashSet<Integer>();
        for(var num : nums)
            hs.add(num);

        var newHead = new ListNode(-1, head);
        var curr = newHead;;
        while(curr.next != null){
            if(hs.contains(curr.next.val)){
                curr.next = curr.next.next;
            }
            else{
                curr = curr.next;
            }
        }
        return newHead.next;
    }
}

/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */

File: 3217-delete-nodes-from-linked-list-present-in-array/3217-delete-nodes-from-linked-list-present-in-array.go
func modifiedList(nums []int, head *ListNode) *ListNode {
    uniq := make(map[int]bool)
    for _, num := range nums {
        uniq[num] = true
    }

    newHead := &ListNode{-1, head}
    curr := newHead

    for curr.Next != nil {
        if uniq[curr.Next.Val] {
            curr.Next = curr.Next.Next
        } else {
            curr = curr.Next
        }
    }
    return newHead.Next
}

/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */